webapp/Connector/src/utility/PlotTooltip.js

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright (c) 2015-2016 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
+ */
+// This is a singleton for charting utilities
+Ext.define('Connector.utility.PlotTooltip', {
+
+    alternateClassName: ['PlotTooltipUtils'],
+
+    singleton: true,
+
+    loadBinTooltipData: function(plot, point, datas, plotName)
+    {
+        var rawRows = this.getRawDataRows(plot, datas);
+        var participantSeqContainerMap = {};
+        Ext.each(rawRows, function(row){
+            var participantSeq = row[QueryUtils.SUBJECT_SEQNUM_ALIAS];
+            participantSeqContainerMap[participantSeq] = row[QueryUtils.CONTAINER_ALIAS];
+        });
+        var participantSeqs = Object.keys(participantSeqContainerMap);
+
+        // in the case when a bin contains > 200 participant, skip querying studies for better perf
```

Comment:
It's ok to skip for >200 since non of the studies on production site has over 150 subjects.